pulp_gem/app/tasks/publishing.py

```diff
@@ -150,16 +151,18 @@ def publish(repository_version_pk):
         os.mkdir("info")
         for name in names_qs:
             lines = []
+            version_list = []
             for gem in gems_qs.filter(name=name):
                 deps = ",".join((f"{key}:{value}" for key, value in gem.dependencies.items()))
-                line = f"{gem.version} {deps}|checksum:{gem.checksum}"
+                line = f"{gem.ext_version} {deps}|checksum:{gem.checksum}"
                 if gem.required_ruby_version:
                     line += f",ruby:{gem.required_ruby_version}"
                 if gem.required_rubygems_version:
                     line += f",rubygems:{gem.required_rubygems_version}"
                 lines.append(line)
+                version_list.append(gem.ext_version)
             info_metadata = _publish_compact_index(lines, f"info/{name}", publication)
-            versions = ",".join(gems_qs.filter(name=name).values_list("version", flat=True))
+            versions = ",".join(version_list)
```

Comment:
Do you have an example gem that I can see to verify this is how the info file should look?

Reply:
This file is in the format:
```
<pgk_name> <version1>,<version2>,...<version5> <old_checksum>
<pkg_name> <another_version1>,<another_version2> <less_old_checksum>
<pkgname> <newest_version> <current_checksum>
```

The idea is that you can just append a new version to it instead of modifying the whole file.
Since we generate a new file anyway, we generate the most compact version of it.

See e.g.:
`http https://rubygems.org/versions | grep pulpcore_client`